app/secrets.py

```diff
@@ -18,5 +16,31 @@
 
 def validate_required_secrets(secrets):
     for required_secret in EXPECTED_SECRETS:
-        if required_secret not in secrets:
+        if required_secret not in secrets['secrets']:
             raise Exception("Missing Secret [{}]".format(required_secret))
+
+
+class SecretStore:
+    def __init__(self, secrets):
+        if 'secrets' in secrets:
+            self.secrets = secrets['secrets']
+        if 'private_keys' in secrets:
+            self.private_keys = secrets['private_keys']
+        if 'public_keys' in secrets:
+            self.public_keys = secrets['public_keys']
+
+    def get_secret_by_name(self, secret_name):
+        if secret_name in self.secrets:
+            return self.secrets[secret_name]
+
+    def get_private_key_by_kid(self, kid):
+        if kid in self.private_keys:
+            return self.private_keys[kid]
+
+        raise InvalidTokenException("Invalid Private Key Identifier [{}]".format(kid))
+
+    def get_public_key_by_kid(self, kid):
+        if kid in self.public_keys:
```

Comment:
I agree. But, Would need to work out how we identify the keys where we only have one version. e.g. how would we know what public key we would use for submission?